src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -154,56 +152,30 @@ private void abortAnalysis() {
 
 
     /**
-     * Sends the requests to Goblint server and reads their results.
+     * Sends the requests to Goblint server and gets their results.
      *
      * @return a collection of warning messages and cfg code lenses if request was successful, null otherwise.
      */
 
     private Collection<AnalysisResult> reanalyse() {
 
-        Request analyzeRequest = new Request("analyze");
-        Request messagesRequest = new Request("messages");
-        Request functionsRequest = new Request("functions");
-
         try {
             // Analyze
             analysisRunning.set(true);
-            AnalyzeResponse analyzeResponse = (AnalyzeResponse) getResponse(analyzeRequest);
+            GoblintAnalysisResult analyzeResult = goblintService.analyze(new Params()).get();
```

Comment:
It works here, but `get` isn't the nicest way to use a `CompletableFuture`. Instead there are methods like `thenAccept` to define callbacks that get run then the results arrive, instead of having a thread be blocked waiting for a `get`.